src/compiler/crystal/compiler.cr

```diff
@@ -559,17 +559,26 @@ module Crystal
       end
 
       def compile
-        bc_name = self.bc_name
-        object_name = self.object_name
+        if compiler.thin_lto
+          compile_to_thin_lto
+        else
+          compile_to_object
+        end
+      end
 
+      private def compile_to_thin_lto
         {% unless LibLLVM::IS_38 || LibLLVM::IS_39 %}
-        if compiler.thin_lto
-          @llvm_mod.write_bitcode_with_summary_to_file(object_name)
+          llvm_mod.write_bitcode_with_summary_to_file(object_name)
           @reused_previous_compilation = false
-          llvm_mod.print_to_file ll_name if compiler.dump_ll?
-          return
-        end
+          dump_llvm_ir
+        {% else %}
+          raise {{ "ThinLTO is not available in LLVM #{LibLLVM::VERSION}".stringify }}
```

Comment:
Would it be easy to raise instantly in the option parser in addition to this, not after several (possibly expensive) compiler passes?

Reply:
Oh never mind, we can't get to this statement if we don't have the correct LLVm version, as we simply compile out the `--lto` option.